### PR TITLE
YLP-1199 Stats in trends view may be wrong

### PR DESCRIPTION
**Pull request by yanngit**

### CODE REVIEW

**Comment by TimCanu** [comment]
Don't forget about the changelog :slightly_smiling_face:

File: packages/viz/test/utils/data.test.js [hunk 1]
```diff
@@ -569,8 +569,16 @@ describe("DataUtil", () => {
       filterAllSpy.restore();
     });
 
-    it("should set the `days` property based on the endpoint range", () => {
-      dataUtil = new DataUtil(smbgData, opts({ endpoints: twoWeekEndpoints, chartPrefs: {} }));
+    it.only("should set the `days` property based on the endpoint range", () => {
```
Comment (TimCanu) [inline packages/viz/test/utils/data.test.js:572]: .only

my man

File: packages/viz/src/utils/data.js [hunk 2]
```diff
@@ -109,7 +108,17 @@ class DataUtil {
...
+      console.log(activeDays);
+
+      // here is a more realistic version of calculating days, fixing a bug in stats
+      this.days = 0;
+      let start = moment(this._endpoints[0]);
+      for(let i = 1; i <= daysInRange; i++) {
+        if (activeDays.includes(start.day())) {
+          this.days += 1;
+        }
+        start.add(1,"d");
```
Comment (yvon-dblg) [inline packages/viz/src/utils/data.js:120]:
```suggestion
        start.add(1, "d");
```

File: packages/viz/test/utils/data.test.js [hunk 3]
```diff
@@ -569,8 +569,16 @@ describe("DataUtil", () => {
...
+      dataUtil = new DataUtil(smbgData, opts({ endpoints: twoWeekEndpoints, chartPrefs: {activeDays: {
+        monday: true,
```
Comment (TimCanu) [inline packages/viz/test/utils/data.test.js:574]: code format has to be incorrect but I guess that we are in viz :cowboy_hat_face:

File: packages/viz/src/utils/data.js [hunk 4]
```diff
@@ -109,7 +108,17 @@ class DataUtil {
 
       this.filter.byActiveDays(activeDays);
 
-      this.days = daysInRange / 7 * activeDays.length;
+      console.log(activeDays);
+
+      // here is a more realistic version of calculating days, fixing a bug in stats
+      this.days = 0;
+      let start = moment(this._endpoints[0]);
```
Comment (yvon-dblg) [inline packages/viz/src/utils/data.js:115]:
```suggestion
      let start = moment.utc(this._endpoints[0]);
```
Or use the timezone. But never use moment() directly

**Review by TimCanu — APPROVED** [review]

**Pull request merged**